dask/array/tests/test_reductions.py

```diff
@@ -809,3 +809,37 @@ def test_nan_func_does_not_warn(func):
     with pytest.warns(None) as rec:
         getattr(da, func)(d).compute()
     assert not rec  # did not warn
+
+
+@pytest.mark.parametrize("chunks", [((3, 3), (2, 1, 2, 1)), ((2, 1, 2, 1), (3, 3))])
+def test_chunk_structure_dependence(chunks):
```

Comment:
It'd be good to have a test that includes more than 2 dimensions. It'd also be good to check with an array that has enough chunks that `split_every` could be tested too.

Reply:
I completely agree.  See the code for a new test implementation (and below for a corresponding example-code).  In fact, after following your suggestion, a few `ValueError` exceptions were raised indicating that the reduction axes in the parameter `axis` had to be supplied in ascending order.  But I've now amended the code in such a way that this requirement no longer matters:

In `_tree_reduce()`, I added the line:
```python
    axis = sorted(axis)
```
just before passing `axis` to `_concatenate2`.